src/main/java/org/terasology/scenario/internal/systems/ConvertIntoEntitySystem.java

```diff
@@ -167,9 +167,14 @@ public void onConvertIntoEntityConstantEvent(ConvertIntoEntityConstantEvent even
 
     @ReceiveEvent
     public void onConvertIntoEntityConstantEvent(ConvertIntoEntityConstantEvent event, EntityRef entity, ScenarioValueRegionComponent component) {
-        for (EntityRef e : entityManager.getEntitiesWith(RegionNameComponent.class)) {
-            if (e.getComponent(NetworkComponent.class).getNetworkId() == Integer.parseInt(event.getValue())) {
-                component.regionEntity = e;
+        if (event.getValue().substring(0,1).equals("x")) { //No network component
+            component.regionEntity = entityManager.getEntity(Integer.parseInt(event.getValue().substring(1)));

```

Comment:
Will fix for longs, this basically is the fix for multiplayer/singleplayer. the original of only using the network id only worked for multiplayer. In single player the entities don't have a network manager setup so all entities have a network id of 0. This means that it needs to use local id for singleplayer(which is fine, because that always matches as there is only one instance of the entity. Multiplayer the way of tracking entities is by using the network id (these always match, in place of the normal ids which don't always match like you said)